src/core/treeLeafValue.ts

```diff
@@ -34,7 +34,9 @@ class _TreeLeafValueBase {
   /**
    * Does the node has a component at the given path. e.g having `routes/users/index.vue` and `routes/users.vue`.
    */
-  filePath?: string
+  filePaths?: {
```

Comment:
I think we should just use a `Map<string, string>`. It should be easier to check if it's empty or not